feat(helm): Add --output flag to version subcommand

vitorfloriano opened the pull request:
This PR introduces the `--output` flag to the `version` subcommand. This flag allows adding more formats to the `version` output without adding additional flags.

It also adds the `go` and `human` formats to the `--output` flag, for printing the version in a raw Go struct and a human-readable table format, respectively, as well as a proper error message:

```
$ helm version -o go
version.BuildInfo{Version:"v4.1+unreleased", GitCommit:"f05c21b6b6380d923696a7684c80a5e0847fbd7b", GitTreeState:"dirty", GoVersion:"go1.25.4", KubeClientVersion:"v1.34"}

$ helm version -o human
Version:             v4.1+unreleased
Git Commit:          f05c21b6b6380d923696a7684c80a5e0847fbd7b
Git Tree State:      dirty
Go Version:          go1.25.4
KubeClient Version:  v1.34

$ helm version -o asdf
Error: invalid output format: "asdf" (valid formats: "go", "human")
```

**Special notes for your reviewer**:
The current output of `version` has not been changed, for backwards compatibility:
```
$ helm version
version.BuildInfo{Version:"v4.1+unreleased", GitCommit:"f05c21b6b6380d923696a7684c80a5e0847fbd7b", GitTreeState:"dirty", GoVersion:"go1.25.4", KubeClientVersion:"v1.34"}
```
We may switch to the human-readable format as the default output for `version` in the next major release.

As follow-up, I suggest the following improvements:
- Update the help message for the version subcommand.
- Add shell completion for the available flags.
- Add other output formats, such as `yaml` and `json`.

**If applicable**:
- [x] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

vitorfloriano commented:
> I am wondering how much this information is parsed by tools and if we should make it behind a `-o` flag or similar.

Hard to know that...but I think having the default output not human-readable is bad UX. The machine-readable output is the one that should be behind a `--json`  or `--verbose` flag or something like that.

OR, we could deprecate the `--short` flag and make it the new default, and make the current default output behind a `--verbose` flag, but that would be disruptive as well. 🤔

vitorfloriano commented:
@banjoh By reading the spec I understand that this would be a breaking change. Thanks for pointing that out. I could change the approach and add a flag for formatted print, but honestly I think this command has enough flags already 😅 What do y'all think?

mrlunchbox777 commented:
@vitorfloriano thank you for asking, for the `-o` arguments idea, in particular the `human-readable` option,  I was thinking that it would be a way to implement your improvement immediately as a flag, then later have it become the default. i.e.:

```yaml
helm 4: # after this MR
  go:
     - "helm version"
     - "helm version -o go"
  humanReadable:
     - "helm version -o human-readable"
  json:
     - "helm version -o json"
  yaml:
     - "helm version -o yaml"
  ...
helm 5:
  go:
     - "helm version -o go"
  humanReadable:
     - "helm version"
     - "helm version -o human-readable"
  json:
     - "helm version -o json"
  yaml:
     - "helm version -o yaml"
   ...
```

To be clear, I just suggested some names that I thought might be good for the options for the arguments, but if there is something better I'd be in favor of that, e.g. `go-format` or `human`

EDIT: I don't think this MR needs to add exhaustive, or even many, output conversions to this. Those could be additional issues.

vitorfloriano committed:
feat(helm): Add --output flag to version subcommand

The version subcommand now includes the --output flag for formatting the version representation. This commit adds the 'go' and 'human' arguments for printing the raw go struct and a human-friendly table format, respectively.

Signed-off-by: vitorfloriano <107767584+vitorfloriano@users.noreply.github.com>

banjoh reviewed (approved):
LGTM
